hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java

```diff
@@ -1502,26 +1503,38 @@ public long submitServerCrash(ServerName serverName, boolean shouldSplitWal) {
     // server state to CRASHED, we will no longer accept the reportRegionStateTransition call from
     // this server. This is used to simplify the implementation for TRSP and SCP, where we can make
     // sure that, the region list fetched by SCP will not be changed any more.
-    serverNode.writeLock().lock();
+    if (serverNode != null) {
```

Comment:
I think here we'd better extract a special method for handling the unknown server case? It will make the code much cleaner, now lots of serverNode != null in the code base which means the code hard to understand...